File: src/com/mark/code/java/Code_771.java
package com.mark.code.java;

/**
 * leetcode 上的771题,宝石与石头,题意就是 S中有多少个J的元素.
 * exp:     1 J= "aA", S = "aAAbbbb" =>3
 *          2 J = "z", S = "ZZ"      =>0
 */
public class Code_771 {

    private int numJewelsInStones(String J,String S) {
        if (J == null || S == null) return 0;
        int sum = 0;
        char[] A = J.toCharArray();
        char[] B = S.toCharArray();

        boolean[] Ab = new boolean[128];
        for (char c : A) {
            Ab[c] = true;
        }
        int cnt = 0;
        for (char c : B) {
            if (Ab[c]) {
                ++cnt;
            }
        }
        return cnt;
    }

}

File: src/com/mark/code/kotlin/Code_771.kt
package com.mark.code.kotlin

/**
 * leetcode 上的771题,宝石与石头,题意就是 S中有多少个J的元素.
 * 我直接用了kotlin自带的函数.
 * 唉. 工作原因,心累.
 * 老哥们看Java那边的改吧
 */
class Code_771 {

    fun numJewelsInStones(J: String, S: String): Int {
        var sum = 0

        S.forEach { v ->
            if (J.find { it == v } != null) {
                sum++
            }
        }

        return sum
    }

}